src/sqlfluff/core/rules/std.py

```diff
@@ -714,25 +717,36 @@ class Rule_L004(BaseCrawler):
         FROM foo
     """
 
-    def _eval(self, segment, raw_stack, memory, **kwargs):
-        """Mixed Tab and Space indentation found in file.
+    config_keywords = ["indent_unit", "tab_space_size"]
 
-        We use the `memory` feature here to keep track of
-        what we've seen in the past.
+    def _eval(self, segment, **kwargs):
+        """Incorrect indentation found in file.
 
+        We only fix tabs to spaces. Spaces to tabs is not clean as the number
+        of space indents might not be a multiple of tab_space_size.
         """
-        indents_seen = memory.get("indents_seen", set())
-        if segment.is_type("whitespace"):
-            if len(raw_stack) == 0 or raw_stack[-1].is_type("newline"):
-                indents_here = set(segment.raw)
-                indents_union = indents_here | indents_seen
-                memory["indents_seen"] = indents_union
-                if len(indents_union) > 1:
-                    # We are seeing an indent we haven't seen before and we've seen others before
-                    return LintResult(anchor=segment, memory=memory)
-                else:
-                    return LintResult(memory=memory)
-        return LintResult(memory=memory)
+        correct_indent = (
+            " " * self.tab_space_size if self.indent_unit == "space" else "\t"
```

Comment:
I think it'd look a little cleaner if you had 2 variables

`whitespace = " "`
and 
`tab = "\t"`

Reviewing the code it's not super natural to read right now ^^